### PR TITLE
SUS-2651 | hitcounter table sunset

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-2651

MySQL 5.7 instances in Reston report the following:

```sql
Warning: "twelverottenfruitschronicles.hitcounter" uses engine "MEMORY" and will not be backed up.
Warning: "twelvesusohfour.hitcounter" uses engine "MEMORY" and will not be backed up.
Warning: "twemtest.hitcounter" uses engine "MEMORY" and will not be backed up.
Warning: "twentiethcenturyclub.hitcounter" uses engine "MEMORY" and will not be backed up.
...
```

`maintenance/tables.sql` has the following entry:

```sql
CREATE TABLE /*_*/hitcounter (
  hc_id int unsigned NOT NULL
) ENGINE=HEAP MAX_ROWS=25000;
```

Let's backport MediaWiki changes that made this feature disappear in MediaWiki 1.25 - https://github.com/wikimedia/mediawiki/commit/90d90dad6e4d463b3db425d78f7ed1e084ca0816 + https://github.com/wikimedia/mediawiki/commit/5f8edb2c0a24b88b84f66b851b63a01b36b1a12c

### From users perspective the following features will be removed:

* Special:PopularPages special page (page views counter was disabled, no meaningful information were presented there), e.g. http://muppet.wikia.com/wiki/Special:PopularPages
* `numberofviews` Parser's magic word

### Per-wiki databases cleanup

* `page.page_counter` column to be dropped
* `site_stats.ss_total_views` column to be dropped
* `hitcounter` table

### `update.php`

We'll run `maintenance/update.php` script site-wide when these changes are deployed:

```
Dropping table hitcounter ...done.
Table site_stats contains ss_total_views field. Dropping... done.
Table page contains page_counter field. Dropping... done.
```

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialPageFactory.php#L167](https://github.com/Wikia/app/blob/ab87422d9efd44b7016c9495c70068fd1f85d819/includes/SpecialPageFactory.php#L167): Remove this unused "$wgDisableCounters" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [ApiQuerySiteinfo.php#L342](https://github.com/Wikia/app/blob/ab87422d9efd44b7016c9495c70068fd1f85d819/includes/api/ApiQuerySiteinfo.php#L342): Remove this unused "$wgDisableCounters" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialStatistics.php#L81](https://github.com/Wikia/app/blob/ab87422d9efd44b7016c9495c70068fd1f85d819/includes/specials/SpecialStatistics.php#L81): Remove this unused "$viewsStats" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialStatistics.php#L84](https://github.com/Wikia/app/blob/ab87422d9efd44b7016c9495c70068fd1f85d819/includes/specials/SpecialStatistics.php#L84): Remove this unused "$wgDisableCounters" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/ab87422d9efd44b7016c9495c70068fd1f85d819/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialPageFactory.php#L167](https://github.com/Wikia/app/blob/a546f047672d262e4a7f0f9d74ccd9d7e8c30a8e/includes/SpecialPageFactory.php#L167): Remove this unused "$wgDisableCounters" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [ApiQuerySiteinfo.php#L342](https://github.com/Wikia/app/blob/a546f047672d262e4a7f0f9d74ccd9d7e8c30a8e/includes/api/ApiQuerySiteinfo.php#L342): Remove this unused "$wgDisableCounters" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialStatistics.php#L81](https://github.com/Wikia/app/blob/a546f047672d262e4a7f0f9d74ccd9d7e8c30a8e/includes/specials/SpecialStatistics.php#L81): Remove this unused "$viewsStats" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [SpecialStatistics.php#L84](https://github.com/Wikia/app/blob/a546f047672d262e4a7f0f9d74ccd9d7e8c30a8e/includes/specials/SpecialStatistics.php#L84): Remove this unused "$wgDisableCounters" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/a546f047672d262e4a7f0f9d74ccd9d7e8c30a8e/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/9c6c630ee143aa9112fad25d400919aa1e0d1f8c/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: includes/DefaultSettings.php [hunk 1]
```diff
@@ -3942,11 +3933,6 @@
  */
 $wgAggregateStatsID = false;
 
-/** Whereas to count the number of time an article is viewed.
- * Does not work if pages are cached (for example with squid).
- */
-$wgDisableCounters = false;
-
```
Comment (macbre) [inline includes/DefaultSettings.php:3949]: This is still set in config repo:

```
CommonSettings.php
33:$wgDisableCounters = true;
```

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/c88c7e249ba013afb16b07b008d4d2c4bb37a0de/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/a74aa3801363a40c0a7934a5a606e62911e6f32d/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/b1ade2e15e6ca1f14130450a90ccbe35fa8c6b0c/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/c85db669c101366f90fd78249b7f65fdf3fc34d5/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/c868aea0e4e621ee83bf6e4e97412f6b3283f7ff/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/604b98f9ecd3879e76e9f587d4a9d9833927179c/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/8b2278c62853fd20b0f725045de05ebdd039ef2e/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/8b2278c62853fd20b0f725045de05ebdd039ef2e/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/dc43a870400c48d1c7d049aecd39a02ccd644334/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [AutoLoader.php#L62](https://github.com/Wikia/app/blob/768d454221b2d97a4bb133451e5049d1b2e816fc/includes/AutoLoader.php#L62): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**